test-data/unit/check-isinstance.test

```diff
@@ -1443,6 +1443,28 @@ else:
 [builtins fixtures/isinstancelist.pyi]
 
 
+[case testIssubclasDestructuringUnions]
+from typing import Union, List, Tuple, Dict, Type
+def f(x: Union[Type[int], Type[str], Type[List]]) -> None:
```

Comment:
I fixed it by choosing one canonical internal representation for both `Union[Type[...]]` and `Type[Union[...]]` . Instances of `TypeType` now have to created using `.make()` static method, which will check if it should return `UnionType` or `TypeType`. It's very reliable because I disallowed the use of `TypeType` constructor (otherwise, we can't prevent someone in the future accidentally creating the wrong internal representation).

As a result, this commit accidentally fixed a broken test, `testTypeUsingTypeCClassMethodUnion`.

Unfortunately, it also means a slight annoyance of having to use a class method to create `TypeType`.

The alternative is for me to renaming class `TypeType` into `_TypeType`, and exposing a function `TypeType`. This will completely keep the current API. But I'm afraid a lot of type hints would break because functions and classes aren't equivalent for purposes of typing, even if they have the same signature). But lmk if you prefer this approach.